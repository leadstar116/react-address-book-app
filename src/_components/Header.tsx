import React from 'react'
import { Link } from 'react-router-dom'
import { Navbar, Nav, FormControl, InputGroup } from 'react-bootstrap'

type Props = {
    withSearchBar: boolean,
    searchString: string,
    setSearchString: (name: string) => void
}

const Header = (props: Props) => {
    return (
        <Navbar bg="light" expand="lg" fixed="top">
            <Navbar.Brand href="#home">Address Book</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                    <Link className="mr-2" to="/">Home</Link>
                    <Link to="/settings">Settings</Link>
                </Nav>
                {props.withSearchBar &&
                <InputGroup className="mr-sm-2 search-input" >
                    <InputGroup.Prepend>
                        <InputGroup.Text className={props.searchString && "border border-danger"}>
                            First and last name
                        </InputGroup.Text>
                    </InputGroup.Prepend>
                    <FormControl
                        type="text"
                        placeholder="Search"
                        className={props.searchString
                            ? "mr-sm-2 border border-danger"
                            : "mr-sm-2"
                        }
                        value={props.searchString}
                        onChange={(e) => props.setSearchString(e.target.value)}
                    />
                </InputGroup>
                }
            </Navbar.Collapse>
        </Navbar>
    )
}

export default Header