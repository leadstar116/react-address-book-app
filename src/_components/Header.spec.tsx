import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header'


describe('Header Component', () => {
    let component = renderer.create(
        <BrowserRouter>
            <Header
                withSearchBar={false}
                searchString=""
                setSearchString={(name: string) => {}}
            />
        </BrowserRouter>
    );

    it('should render with given props', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})