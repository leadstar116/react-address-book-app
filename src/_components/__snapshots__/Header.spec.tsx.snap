// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render with given props 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light fixed-top"
>
  <a
    className="navbar-brand"
    href="#home"
  >
    Address Book
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="mr-2"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
</nav>
`;
